libraries/psio/include/psio/reflect.hpp

```diff
@@ -0,0 +1,760 @@
+#pragma once
+#include <map>
+#include <memory>
+#include <optional>
+#include <span>
+#include <string_view>
+#include <tuple>
+#include <type_traits>
+#include <variant>
+#include <vector>
+
+#include <boost/core/demangle.hpp>
+#include <boost/preprocessor/cat.hpp>
+#include <boost/preprocessor/control/iif.hpp>
+#include <boost/preprocessor/facilities/check_empty.hpp>
+#include <boost/preprocessor/logical/bitand.hpp>
+#include <boost/preprocessor/logical/compl.hpp>
+#include <boost/preprocessor/seq/filter.hpp>
+#include <boost/preprocessor/seq/for_each.hpp>
+#include <boost/preprocessor/seq/for_each_i.hpp>
+#include <boost/preprocessor/seq/to_tuple.hpp>
+#include <boost/preprocessor/seq/transform.hpp>
+#include <boost/preprocessor/stringize.hpp>
+#include <boost/preprocessor/tuple/push_front.hpp>
+#include <boost/preprocessor/tuple/rem.hpp>
+#include <boost/preprocessor/variadic/to_seq.hpp>
+
+#include <psio/get_type_name.hpp>
+#include <psio/member_proxy.hpp>
+
+namespace psio
+{
+   template <typename R, typename C, typename... Args>
+   constexpr std::tuple<std::decay_t<Args>...> args_as_tuple(R (C::*)(Args...));  // { return ; }
+   template <typename R, typename C, typename... Args>
+   constexpr std::tuple<std::decay_t<Args>...> args_as_tuple(R (C::*)(Args...) const)
+   {
+      return {};
+   }
+   template <typename R, typename... Args>
+   constexpr std::tuple<std::decay_t<Args>...> args_as_tuple(R (*)(Args...))
+   {
+      return {};
+   }
```

Comment:
Should use `remove_cvref_t` instead of `decay_t`.  Array-to-pointer decay is definitely wrong here.